File: org.palladiosimulator.blockchainsystems.core/src/main/java/org/palladiosimulator/blockchainsystems/core/blocks/BlockImpl.java
package org.palladiosimulator.blockchainsystems.core.blocks;

import java.util.HashSet;

import org.palladiosimulator.blockchainsystems.core.system.abstractions.Block;

public record BlockImpl(
		String hash,
		String previousHash,
		String originId,
		long blockMinedTimesteamp,
		int size,
		HashSet<String> tags) implements Block {

	@Override
	public String getHash() {
		return this.hash();
	}

	@Override
	public String getPreviousHash() {
		return this.previousHash();
	}

	@Override
	public String getOriginId() {
		return this.originId();
	}

	@Override
	public long getBlockMinedTimestamp() {
		return this.blockMinedTimesteamp();
	}
	
	@Override
	public int getSize() {
		return size;
	}

	@Override
	public boolean hasTag(String tag) {
		return tags.contains(tag);
	}

	
}

File: org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/block/BlockImpl.kt
package org.palladiosimulator.blockchainsystems.core.block

import org.palladiosimulator.blockchainsystems.core.block.abstractions.Block
import org.palladiosimulator.blockchainsystems.core.transaction.abstractions.Transaction

open class BlockImpl(
  override val hash: String,
  override val previousHash: String?,
  override val originId: String?,
  override val blockMinedTimestamp: Long,
  override val size: Int,
  override val transactions: Set<Transaction>,
  val tags: Set<String>
) : Block {
  override fun hasTag(tag: String): Boolean {
    return tags.contains(tag)
  }
}